Matching

BrandonWade opened the pull request:
Adds additional fields to the registration page needed for user matching

BrandonWade committed:
Added fields to registration page & layout changes

BrandonWade committed:
Updated styling of radius drop down.

BrandonWade committed:
More fixes to user page layout

BrandonWade committed:
User radius now works with DB

BrandonWade committed:
Fixed merge conflicts with angular-ui-router code

paymog commented:
Need to write tests.

paymog committed:
Wrote tests

paymog committed:
Can now match users. There's no alg, just matching

When two separate users decide they'd like to be matched, they will be
appropriately notified using socket.io.

Other changes include:
- cookie now stores a jsonified object
- register and authenticate api return the full user object w/o password
- fixed enter button on register page
- added new field to users page (not tested because we're moving to josh's home page)

paymog commented:
I specifically didn't test the change to UserController because we're moving away from it with `LoginLanding`

kyleleung75 commented:
Everything looked good through the files, I am still going to run it locally and make sure everything is good.

paymog commented:
Sounds good

On Wed, Mar 11, 2015, 10:30 PM kyleleung75 notifications@github.com wrote:

> Everything looked good through the files, I am still going to run it
> locally and make sure everything is good.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/paymog/Lunchr/pull/40#issuecomment-78417976.

kyleleung75 commented:
AHH the matching is beautiful! It resets the user too when you reclick the matching button. The tests were good too. I suppose we should think about making a timeout for it so that it can send back a message for them to try again later, but everything looks great!

paymog commented:
Sick. The only problem is refreshing the page at any point after you've
started the matching process. I tried to come up with a good solution but
failed. However, I think I've though of one since I put up the PR. I'll
implement that once the landing page goes through.

On Wed, Mar 11, 2015 at 11:10 PM kyleleung75 notifications@github.com
wrote:

> AHH the matching is beautiful! It resets the user too when you reclick the
> matching button. The tests were good too. I suppose we should think about
> making a timeout for it so that it can send back a message for them to try
> again later, but everything looks great!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/paymog/Lunchr/pull/40#issuecomment-78420963.

kjrcda commented:
Okay cool. I should have it done tomorrow. I may have to redo my changes on a new branch. rebasing now is a ****\* because I've done it 3 times already. I'll evaluate tomorrow which is quicker (rebate or new branch) and get it done ASAP. Shouldn't take more than 1hr either way.

-------- Original message --------
From: Paymahn Moghadasian notifications@github.com
Date:03-11-2015  11:30 PM  (GMT-06:00)
To: paymog/Lunchr Lunchr@noreply.github.com
Cc:
Subject: Re: [Lunchr] Matching (#40)

Sick. The only problem is refreshing the page at any point after you've
started the matching process. I tried to come up with a good solution but
failed. However, I think I've though of one since I put up the PR. I'll
implement that once the landing page goes through.

On Wed, Mar 11, 2015 at 11:10 PM kyleleung75 notifications@github.com
wrote:

> AHH the matching is beautiful! It resets the user too when you reclick the
> matching button. The tests were good too. I suppose we should think about
> making a timeout for it so that it can send back a message for them to try
> again later, but everything looks great!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/paymog/Lunchr/pull/40#issuecomment-78420963.

---

Reply to this email directly or view it on GitHub:
https://github.com/paymog/Lunchr/pull/40#issuecomment-78422296

paymog committed:
Shit got done.

Here's a list:
1) Cookie is updated whenever db model changes
2) home page controller uses updated auth service
3) home page controller correctly navigates after page refreshes
4) Tests. Lots of fucking tests.
5) Socket emits complete user object when save succeeds

kjrcda commented on webapp/ng_tests/controllers/home_matching_controller_test.js:
```diff
@@ -42,17 +42,27 @@ describe('UserMatchingController', function () {
+            socket.receive('hasBeenMatched', userResponse);
+
+            expect(authService.setUser).toHaveBeenCalledWith(userResponse.user)
         })
```
;

kjrcda commented on webapp/ng_tests/controllers/home_matched_controller_test.js:
```diff
@@ -0,0 +1,35 @@
+
+            expect(authService.currentUser).toHaveBeenCalled();
+            expect($rootScope.name).toBe(FIRST_NAME);
+        })
```
;

kjrcda commented on webapp/ng_tests/controllers/home_page_controller_test.js:
```diff
@@ -23,27 +23,42 @@ describe('HomePageController', function () {
+            $state.expectTransitionTo('home.matching');
+
+            createController(authService);
         })
```
;

kjrcda commented on webapp/ng_tests/controllers/main_page_controller_test.js:
```diff
@@ -90,7 +94,7 @@ describe('MainPageController', function () {
             expect($rootScope.errorMessages).toBeFalsy();
-            expect(authService.login.calls.any()).toBeFalsy();
+            expect(authService.setUser.calls.any()).toBeFalsy();
         })
```
;

kjrcda commented on webapp/ng_tests/controllers/main_page_controller_test.js:
```diff
@@ -100,7 +104,7 @@ describe('MainPageController', function () {
             $rootScope.createAccount();
-            expect(authService.login.calls.any()).toBeFalsy();
+            expect(authService.setUser.calls.any()).toBeFalsy();
         })
```
; and line below

kjrcda commented on webapp/ng_tests/controllers/register_controller_test.js:
```diff
@@ -129,7 +165,7 @@ describe('RegisterController', function () {
             expect($rootScope.errorMessages).toBe(errorMessage);
-            expect(authService.login.calls.any()).toBeFalsy();
+            expect(authService.setUser.calls.any()).toBeFalsy();
         })
```
;

kjrcda commented on webapp/public/javascripts/controllers.js:
```diff
@@ -67,31 +72,44 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
+        socket.on('matched' + currentUser.email, function (data) {
+            var userObject = angular.fromJson(data.user);
+            authService.setUser(userObject);
+            $state.go('home.matched')
```
;

kjrcda commented on webapp/public/javascripts/controllers.js:
```diff
@@ -67,31 +72,44 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
-            });
+        var currentUser = authService.currentUser();
+        if(currentUser.matchedWith){
+            $state.go('home.matched')
```
;

kjrcda commented on webapp/public/javascripts/controllers.js:
```diff
@@ -67,31 +72,44 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
+            $state.go('home.matched')
+        }
+        if(currentUser.wantsToBeMatched){
+            $state.go('home.matching')
```
;

kjrcda commented on webapp/public/stylesheets/style.css:
```diff
@@ -22,7 +22,29 @@ a {
+.radiusButton {
+    width: 100px;
+    text-align: left;
+}.div-center-circleBtn {
```
newline? weird how that happened lol

kjrcda commented on webapp/routes/api.js:
```diff
@@ -36,29 +37,31 @@ router.post('/users/register', function (req, res, next) {
             email: req.body.email, password: req.body.password,
-            firstname: req.body.firstname, lastname: req.body.lastname
-        })
+            firstname: req.body.firstname, lastname: req.body.lastname,
```
maybe put all fields on separate lines

kjrcda commented on webapp/routes/api.js:
```diff
@@ -36,29 +37,31 @@ router.post('/users/register', function (req, res, next) {
 
         user.save(function (error, user) {
             if (error) {
                 return next(error)
```
;

kjrcda commented on webapp/routes/socket.js:
```diff
@@ -1,8 +1,68 @@
+        var currentUser = null;
+        User.findOne({email: data.userEmail}, function(error, user) {
+            if(error){
+                return
```
;

kjrcda commented on webapp/routes/socket.js:
```diff
@@ -1,8 +1,68 @@
+                    return;
+                }
+                currentUser.password = null;
+                socket.emit('hasBeenMatched', {user: JSON.stringify(currentUser)})
```
;

kjrcda commented on webapp/routes/socket.js:
```diff
@@ -1,8 +1,68 @@
+        User.find({email: {'$ne': data.userEmail }, wantsToBeMatched: true}, function(error, users){
+            if(error || users.length == 0){
+                //bleh
+                return
```
; and remove comment

kjrcda commented on webapp/views/partials/home.jade:
```diff
@@ -5,4 +5,6 @@ div(class='container')
-        button(type='button', class='btn btn-block btn-primary', ng-click='editInfo()') Edit my Information
\ No newline at end of file
+        button(type='button', class='btn btn-block btn-primary', ng-click='editInfo()') Edit my Information
+
```
move that comment pertaining to ui-view from other file

kyleleung75 commented:
Everything looked good, just testing it locally.
